pkg/platform/runtime/executor/init.go

```diff
@@ -123,24 +132,44 @@ func (f *Executor) Cleanup() error {
 	return nil
 }
 
-func (f *Executor) createExecutor(targeter Targeter, sockPath string, env map[string]string, exe string) error {
-	name := NameForExe(filepath.Base(exe))
-	target := filepath.Clean(filepath.Join(f.executorPath, name))
+func IsExecutor(filePath string) (bool, error) {
+	if fileutils.IsDir(filePath) {
+		return false, nil
+	}
+
+	b, err := fileutils.ReadFile(filePath)
+	if err != nil {
+		return false, locale.WrapError(err, "err_cleanexecutor_noread", "Could not read potential executor file: {{.V0}}.", filePath)
+	}
+	return isOwnedByUs(b), nil
+}
+
+// executorDenoter is an old denoter that we want to make sure we clean up
+const executorDenoter = "!DO NOT EDIT! State Tool Executor !DO NOT EDIT!"
+
+// shimDenoter is an old denoter that we want to make sure we clean up
+const shimDenoter = "!DO NOT EDIT! State Tool Shim !DO NOT EDIT!"
+
+func isOwnedByUs(fileContents []byte) bool {
+	return strings.Contains(string(fileContents), "state-exec") ||
+		execmeta.IsMetaFile(fileContents) ||
+		// deprecated
+		strings.Contains(string(fileContents), executorDenoter) ||
+		strings.Contains(string(fileContents), shimDenoter)
+}
+
+func copyExecutor(destDir, exe, srcExec string) error {
+	name := filepath.Base(exe)
+	target := filepath.Clean(filepath.Join(destDir, name))
```

Comment:
This appears to be old code that is no longer used.

Reply:
I am not sure how changing this will affect users if they update from an older version which used the denoter messages, so I opted to include them. Otherwise, I'm fairly sure that most of the code is still in use.

Reply:
I don't follow. Whether or not they used this is irrelevant, no code is using any of this anymore. It's dead code, we can drop it.

Reply:
I might be wrong, but... If we don't detect old executors, then the new executors will error out on install. Detecting that they are files under our control allows us to be free to remove the old ones in favor of the new ones.

Reply:
My bad, I though this wasn't being used at all.

Can we put this somewhere that makes it clear that this is legacy code? And rename any external invocations to have a `Legacy()` suffix? Want to make sure we can drop this in the future.